api/permissions.py

```diff
@@ -1,6 +1,8 @@
 from oauth2_provider.ext.rest_framework import TokenHasScope
```

Comment:
Imports should be ordered alphabetically and separated into stdlib, 3rdparty, and application.

```python
import abc
import os

import django
import requests

import share.models
```

Looks like this file is a bit messed up already, would you mind reordering it?